doc/markdownlinkconverter/markdownlinkconverter.c

```diff
@@ -394,34 +400,50 @@ static void printTarget(FILE * output, char * target, char * filenameInElektra)
 			++folderBack;
 			target = target + 3;
 		}
-		// go folderBack times upwards in filenameInElektra
-		char * lastFolderDelimiter = strrchr (&copyFilenameData[1], FOLDER_DELIMITER);
+		// reduce pathToLink folderBack times
+		char * lastFolderDelimiter = strrchr (pathToLink, FOLDER_DELIMITER);
 		if (lastFolderDelimiter != NULL)
 		{
-			*lastFolderDelimiter = '\0';
-			// not in root so target needs to be extended with path
 			while (folderBack>0)
 			{
+				*lastFolderDelimiter = '\0';
 				--folderBack;
-				lastFolderDelimiter = strrchr (copyFilename, FOLDER_DELIMITER);
+				lastFolderDelimiter = strrchr (pathToLink, FOLDER_DELIMITER);
 				if (lastFolderDelimiter == NULL)
 				{
-					fprintf (output, "INVALID LINK");
+					fprintf (stderr, "INVALID LINK %s:%i    \"%s\"\n", inputFilename, lineCount, backupTarget);
 					return;
 				}
-				*lastFolderDelimiter = '\0';
-			}
-			// if copyFilename was not reduced print it
-			if (copyFilename[0] != '\0')
-			{
-				printConvertedPath (output, &copyFilename[1]);
-				fprintf (output, "_");
 			}
 		}
+		strcpy (++lastFolderDelimiter, target);
 	} else {
 		++target; // remove starting /
+		strcpy (&pathToLink[indexofElektraRoot], target);
+	}
+	// if target ends with /, link it to README.md
+	if (target[strlen (target) - 1] == '/')
+	{
+		strcpy (&pathToLink[strlen (pathToLink)], "README.md");
+	}
+	//validate link
+	FILE * test = fopen (pathToLink, "r");
+	if (!test)
+	{
+		fprintf (stderr, "INVALID LINK %s:%i    \"%s\"\n", inputFilename, lineCount, backupTarget);
```

Comment:
Ok, I tested it, it does not work.

E.g. if gcc prints to stderr:
```
/e/src/plugins/ini/ini.c|500 col 13| warning: ‘outputDebug’ defined but not used [-Wunused-function]
```
my editor recognize that the warning is in line 500, column 13. Thus I can directly jump to that point.
The INVALID LINK... does not match the pattern my editor uses, thus it is printed, but the editor cannot jump to the place.

Is it possible that you also print the invalid links in such a format:
```
/e/src/plugins/ini/ini.c|500 col 13| warning: invalid link
```